### ISSUE TITLE
🐛 Bug Report: TechDocsAddons breaks rendering Tech Docs pages after updating to 1.8

### ISSUE THREAD

**Issue by ciprianna** [backstage/backstage#15126]
### 📜 Description

Updated from version 1.7.0 -> 1.8.3 using `yarn backstage-cli versions:bump`. After the upgrade, all tech docs pages failed to render.

The `<TechDocsAddons>` component from `@backstage/plugin-techdocs-react` breaks rendering of tech docs pages. There is no error output in the browser console or logs, as if perhaps there's a null check that doesn't render if data is missing. If the section is commented out, the page loads as expected. This behavior only occurs when children are nested underneath the component. i.e.,

```js
      <TechDocsAddons>
        <ReportIssue />
      </TechDocsAddons>
```

I also tried rendering the Mermaid addon underneath and got the same behavior. If you leave it without children, nothing breaks.

A search through Discord showed other people experiencing similar behavior as early as 1.8.2. I'm trying to bump to 1.8.3, but I'm not sure which specific version is causing the problem.

### 👍 Expected behavior

TechDocsAddons component can be used on TechDocs pages without rendering issues.

### 👎 Actual Behavior with Screenshots

Page fails to render any content:

<img width="1270" alt="Screen Shot 2022-12-08 at 2 10 54 PM" src="https://user-images.githubusercontent.com/8051967/206558819-61cd1271-f546-4782-85ac-9d1540cc476a.png">

No console errors or logs indicating anything is wrong.

### 👟 Reproduction steps

Have a project that is using the `TechDocsAddons` component when rendering TechDocs on version `1.7.*`. Use any addon as a child.

Upgrade backstage using the cli command `yarn backstage-cli versions:bump` to bring everything up to the latest version.

View TechDocs pages.

### 📃 Provide the context for the Bug.

Removing TechDocsAddons works as a temporary work-around, but I'd like to continue using addons. I am waiting on the upgrade for my project until we can fix.

### 🖥️ Your Environment

OS:   Darwin 21.6.0 - darwin/x64
node: v16.18.1
yarn: 1.22.0
cli:  0.21.1 (installed)
backstage:  1.8.3

Dependencies:
  @backstage/app-defaults                          1.0.8
  @backstage/backend-common                        0.15.2, 0.16.0
  @backstage/backend-plugin-api                    0.1.4
  @backstage/backend-tasks                         0.3.7
  @backstage/catalog-client                        1.1.2
  @backstage/catalog-model                         1.1.3
  @backstage/cli-common                            0.1.10
  @backstage/cli                                   0.21.1
  @backstage/config-loader                         1.1.6
  @backstage/config                                1.0.4
  @backstage/core-app-api                          1.2.0
  @backstage/core-components                       0.12.0, 0.9.5
  @backstage/core-plugin-api                       1.1.0
  @backstage/dev-utils                             1.0.8
  @backstage/errors                                1.1.3
  @backstage/integration-react                     1.1.6
  @backstage/integration                           1.4.0
  @backstage/plugin-adr-backend                    0.2.3
  @backstage/plugin-adr-common                     0.2.3
  @backstage/plugin-adr                            0.2.3
  @backstage/plugin-api-docs                       0.8.11
  @backstage/plugin-app-backend                    0.3.38
  @backstage/plugin-auth-backend                   0.17.1
  @backstage/plugin-auth-node                      0.2.7
  @backstage/plugin-catalog-backend-module-github  0.2.1
  @backstage/plugin-catalog-backend                1.5.1
  @backstage/plugin-catalog-common                 1.0.8
  @backstage/plugin-catalog-graph                  0.2.23
  @backstage/plugin-catalog-import                 0.9.1
  @backstage/plugin-catalog-node                   1.2.1
  @backstage/plugin-catalog-react                  1.2.1
  @backstage/plugin-catalog                        1.6.1
  @backstage/plugin-cloudbuild                     0.3.11
  @backstage/plugin-github-actions                 0.5.11
  @backstage/plugin-home                           0.4.27
  @backstage/plugin-org                            0.6.1
  @backstage/plugin-permission-common              0.7.1
  @backstage/plugin-permission-node                0.7.1
  @backstage/plugin-permission-react               0.4.7
  @backstage/plugin-proxy-backend                  0.2.32
  @backstage/plugin-scaffolder-backend             1.8.0
  @backstage/plugin-scaffolder-common              1.2.2
  @backstage/plugin-scaffolder                     1.8.0
  @backstage/plugin-search-backend-module-pg       0.4.2
  @backstage/plugin-search-backend-node            1.0.4
  @backstage/plugin-search-backend                 1.1.1
  @backstage/plugin-search-common                  1.1.1
  @backstage/plugin-search-react                   1.2.1
  @backstage/plugin-search                         1.0.4
  @backstage/plugin-stack-overflow                 0.1.7
  @backstage/plugin-tech-radar                     0.5.18
  @backstage/plugin-techdocs-backend               1.4.1
  @backstage/plugin-techdocs-module-addons-contrib 1.0.6
  @backstage/plugin-techdocs-node                  1.4.2
  @backstage/plugin-techdocs-react                 1.0.6
  @backstage/plugin-techdocs                       1.4.0
  @backstage/plugin-user-settings                  0.5.1
  @backstage/release-manifests                     0.0.7
  @backstage/test-utils                            1.2.2
  @backstage/theme                                 0.2.16
  @backstage/types                                 1.0.1
  @backstage/version-bridge                        1.0.2

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by Rugvip** [comment]
@realandersn @agentbellnorm, would this be related to #14323?

**Comment by dmckernanacv** [comment]
I am seeing this as well, both running in docker locally on macOS and deployed to a kubernetes cluster in EKS.

**Comment by agentbellnorm** [comment]
@Rugvip Possibly! We're looking into it right now.

**Comment by ottosichert** [comment]
Hey there @ciprianna! 👋 

Thanks for reporting, we are currently trying to reproduce the issue. There are multiple ways on how you can set up TechDocs for you Backstage instance, and we would need to know how your setup roughly looks like.

You mentioned this snippet above:

```jsx
<TechDocsAddons>
  <ReportIssue />
</TechDocsAddons>
```

What `<Route>` are you using around it? And what `element` and `children` props does it have?

**Comment by ciprianna** [comment]
Hello, @ottosichert  - thanks for getting back to me on this! Appreciate you all looking into it.

Here's the full snippet:
```
<FlatRoutes>

  <Route
    path="/docs/:namespace/:kind/:name/*"
    element={<TechDocsReaderPage />}
  >
    <TechDocsAddons>
      <ReportIssue />
      <Mermaid config={{ theme: 'forest', themeVariables: { lineColor: '#000000' } }} />
    </TechDocsAddons>
  </Route>

</FlatRoutes>
```

**Comment by ivanaguilario** [comment]
I can confirm this is happening in 1.8.2 and 1.8.3 using the same setup as @ciprianna.

Removing the `<Mermaid config={{ theme: 'forest', themeVariables: { lineColor: '#000000' } }} />` component renders the docs correctly, but some JS errors are thrown in console:

```
Material-UI: The `fade` color utility was renamed to `alpha` to better describe its functionality.

You should use `import { alpha } from '@material-ui/core/styles'`
```

And: 

```
react-dom.development.js:67 Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
    at div
    at Grid (webpack-internal:///../../node_modules/@material-ui/core/esm/Grid/Grid.js:235:35)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at div
    at Grid (webpack-internal:///../../node_modules/@material-ui/core/esm/Grid/Grid.js:235:35)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at p
    at Typography (webpack-internal:///../../node_modules/@material-ui/core/esm/Typography/Typography.js:165:28)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at HeaderLabelContent (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:6315:31)
    at a
    at Typography (webpack-internal:///../../node_modules/@material-ui/core/esm/Typography/Typography.js:165:28)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at Link (webpack-internal:///../../node_modules/@material-ui/core/esm/Link/Link.js:85:23)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at eval (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:545:6)
    at span
    at div
    at Grid (webpack-internal:///../../node_modules/@material-ui/core/esm/Grid/Grid.js:235:35)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at HeaderLabel (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:6319:11)
    at div
    at Grid (webpack-internal:///../../node_modules/@material-ui/core/esm/Grid/Grid.js:235:35)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at header
    at Header (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:6201:5)
    at TechDocsReaderPageHeader (webpack-internal:///../../node_modules/@backstage/plugin-techdocs/dist/esm/index-847211c1.esm.js:1955:11)
    at main
    at ThemeProvider (webpack-internal:///../../node_modules/@material-ui/styles/esm/ThemeProvider/ThemeProvider.js:41:24)
    at Page (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:6411:11)
    at TechDocsReaderLayout (webpack-internal:///../../node_modules/@backstage/plugin-techdocs/dist/esm/index-847211c1.esm.js:2090:3)
    at AnalyticsContext (webpack-internal:///../../node_modules/@backstage/core-plugin-api/dist/index.esm.js:71:11)
    at eval (webpack-internal:///../../node_modules/@backstage/plugin-techdocs-react/dist/index.esm.js:147:6)
    at TechDocsReaderPage$1 (webpack-internal:///../../node_modules/@backstage/plugin-techdocs/dist/esm/index-847211c1.esm.js:2100:114)
    at RoutableExtensionWrapper (webpack-internal:///../../node_modules/@backstage/core-plugin-api/dist/index.esm.js:536:15)
    at PluginProvider (webpack-internal:///../../node_modules/@backstage/core-plugin-api/dist/index.esm.js:297:11)
    at AnalyticsContext (webpack-internal:///../../node_modules/@backstage/core-plugin-api/dist/index.esm.js:71:11)
    at PluginErrorBoundary (webpack-internal:///../../node_modules/@backstage/core-plugin-api/dist/index.esm.js:504:5)
    at Suspense
    at Result (webpack-internal:///../../node_modules/@backstage/core-plugin-api/dist/index.esm.js:599:21)
    at D (webpack-internal:///../../node_modules/react-router/index.js:35:685)
    at FlatRoutes (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2904:81)
    at div
    at SidebarPinStateProvider (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:3662:11)
    at SidebarPage (webpack-internal:///../../node_modules/@backstage/core-components/dist/index.esm.js:3711:82)
    at Root (webpack-internal:///./src/components/Root/Root.tsx:108:17)
    at D (webpack-internal:///../../node_modules/react-router/index.js:35:685)
    at E (webpack-internal:///../../node_modules/react-router/index.js:38:820)
    at SignInPageWrapper (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2739:18)
    at x (webpack-internal:///../../node_modules/react-router/index.js:37:23)
    at w (webpack-internal:///../../node_modules/react-router-dom/index.js:42:23)
    at AppRouter (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2755:26)
    at RoutingProvider (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2201:3)
    at CssBaseline (webpack-internal:///../../node_modules/@material-ui/core/esm/CssBaseline/CssBaseline.js:66:31)
    at WithStyles (webpack-internal:///../../node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at ThemeProvider (webpack-internal:///../../node_modules/@material-ui/styles/esm/ThemeProvider/ThemeProvider.js:41:24)
    at Provider (webpack-internal:///../../node_modules/@backstage/app-defaults/dist/index.esm.js:329:18)
    at AppThemeProvider (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2444:29)
    at AppContextProvider (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2322:3)
    at ApiProvider (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:77:11)
    at Provider (webpack-internal:///../../node_modules/@backstage/core-app-api/dist/index.esm.js:2648:25)
    at App
```

**Comment by ciprianna** [comment]
@ivanaguilario  - I experience this issue trying to render any add-on. If I remove Mermaid and leave it with just `<ReportIssue />` the page fails to render.

**Comment by ivanaguilario** [comment]
@ciprianna can confirm it only works like this:

```
    <TechDocsAddons>
    </TechDocsAddons>
```

**Comment by ciprianna** [comment]
Yes, @ivanaguilario  - same behavior for me. It renders correctly without any children. 👍🏻

**Comment by ivanaguilario** [comment]
@dmckernanacv Yes, I can confirm it works when rendering the docs in the Catalog with the plugin enabled

**Comment by ciprianna** [comment]
@dmckernanacv  - same for me. It renders in the Catalog Docs tab as expected.

**Comment by omartheironman** [comment]
this happened to me as well

**Comment by omartheironman** [comment]
![image](https://user-images.githubusercontent.com/58672497/207762352-7b4e38f7-2821-4355-b006-2c0c2accf353.png)

if its any help, the issue happens right here where we seem to be stuck in a loop for sometime until the it we get the error

index.esm.js:290 Uncaught TypeError: Cannot read properties of undefined (reading 'getSelection')
    at HTMLDocument.<anonymous> (index.esm.js:290:43)
    at H (debounce.js:95:19)
    at un (debounce.js:144:14)
    at J (debounce.js:132:14)

**Comment by ottosichert** [comment]
Quick update:

We haven't been able to reproduce issue yet with the TechDocs configuration above. We set up a test repository with the steps below, and we would greatly appreciate if one of you who are affected can provide us with either an example repository where the issue surfaces, or fork the test repository:

> https://github.com/agentbellnorm/15126-repro-stage

Our steps trying to reproduce:
- `npx @backstage/create-app`
- Already contains TechDocs route structure like above (just without `<MermaidAddon>`)
- `yarn backstage-cli info` => Backstage 1.8.0
- `yarn dev` => TechDocs renders correcty ✅ 
- `backstage-cli versions:bump --release 1.8.3`
- `yarn dev` => TechDocs **still** renders correcty ✅

**Comment by agentbellnorm** [comment]
Just to add to @ottosichert 's comment, we also
* Installed `backstage-plugin-techdocs-addon-mermaid@^0.4.2`
* Added this documentation component: https://github.com/realandersn/backstage-docs-test/blob/master/component-info.yaml
* Added the addon to the addons section in `App.tsx` [link](https://github.com/agentbellnorm/15126-repro-stage/blob/main/packages/app/src/App.tsx#L77)

**Comment by omartheironman** [comment]
> Quick update:
> 
> We haven't been able to reproduce issue yet with the TechDocs configuration above. We set up a test repository with the steps below, and we would greatly appreciate if one of you who are affected can provide us with either an example repository where the issue surfaces, or fork the test repository:
> 
> > https://github.com/agentbellnorm/15126-repro-stage
> 
> Our steps trying to reproduce:
> 
> * `npx @backstage/create-app`
> * Already contains TechDocs route structure like above (just without `<MermaidAddon>`)
> * `yarn backstage-cli info` => Backstage 1.8.0
> * `yarn dev` => TechDocs renders correcty ✅
> * `backstage-cli versions:bump --release 1.8.3`
> * `yarn dev` => TechDocs **still** renders correcty ✅

@ottosichert  just to confirm you also have the reportissue add-on enabled right?

```
<TechDocsAddons>
  <ReportIssue />
</TechDocsAddons> 
```

**Comment by omartheironman** [comment]
for everyone stuck on this, I managed to resolve this by comparing and upgrading my dependancy in package.json 

   "@backstage/plugin-techdocs": "^1.4.0",
    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.6",
    "@backstage/plugin-techdocs-react": "^1.0.6",
     "react-router": "6.3.0",
    "react-router-dom": "6.3.0"
    
    It rendered for me

**Comment by dmckernanacv** [comment]
> for everyone stuck on this, I managed to resolve this by comparing and upgrading my dependancy in package.json
> 
> "@backstage/plugin-techdocs": "^1.4.0", "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.6", "@backstage/plugin-techdocs-react": "^1.0.6", "react-router": "6.3.0", "react-router-dom": "6.3.0"
> 
> ```
> It rendered for me 
> ```

Looks like that fixed it for us too. 
Not sure how that dependency update was missed, we use the command line tool for generating upgrades, looks like it was updated with version [1.7.0](https://backstage.github.io/upgrade-helper/?from=1.6.0&to=1.7.0).

I just checked our PR for upgrading to `1.7.0` and sure enough, `react-router` and `react-router-dom` had no changes.

**Comment by agentbellnorm** [comment]
@dmckernanacv @omartheironman  Ok, yes we have some code in the more recent versions of `@backstage/plugin-techdocs` to handle react router v6. I don't expect it to work with older versions.

What versions of `react-router` and ´react-router-dom´ did you use before you bumped today?

**Comment by dmckernanacv** [comment]
Mine was on 
```
    "react-router": "6.0.0-beta.0",
    "react-router-dom": "6.0.0-beta.0",
```
Which seems to have been used prior to backstage 1.7.0

**Comment by omartheironman** [comment]
mine was set to   
    "react-router": "6.0.0-beta.0",
    "react-router-dom": "6.0.0-beta.0",
    
    
as well

**Comment by omartheironman** [comment]
<img width="1051" alt="image" src="https://user-images.githubusercontent.com/58672497/207895902-4e67b679-798a-4cd0-95f5-9b94a2ae7931.png">

@agentbellnorm @dmckernanacv  I am also seeing this when I open a document also when I select a text nothing happens

**Comment by dmckernanacv** [comment]
> <img alt="image" width="1051" src="https://user-images.githubusercontent.com/58672497/207895902-4e67b679-798a-4cd0-95f5-9b94a2ae7931.png">
> 
> @agentbellnorm @dmckernanacv I am also seeing this when I open a document also when I select a text nothing happens

I'm not seeing this issue

**Comment by omartheironman** [comment]
@agentbellnorm  should I open another issue for this?

**Comment by ciprianna** [comment]
Yes, ours is on

```json
    "react-router": "6.0.0-beta.0",
    "react-router-dom": "6.0.0-beta.0",
```

When we upgrade to latest, everything is working as expected 🎉 

```json
    "react-router": "^6.4.5",
    "react-router-dom": "^6.4.5",
```

**Comment by ciprianna** [comment]
I'll close this since react-router updates resolve the issue. Thanks so much for digging into it!

**Comment by agentbellnorm** [comment]
I think those warnings only show when running locally. They should definitely be fixed, but I'm fairly certain they're not relevant to this issue.

@omartheironman 
> @agentbellnorm should I open another issue for this?

Please do! 💯 

Regarding the versions you both referred to, They're technically supported since it's in the version range of the peer dependencies. But I have been able to confirm [in the test-repo we created](https://github.com/agentbellnorm/15126-repro-stage/commit/7b505c8f830e74b17e784d241afcd243706e3bd2) that downgrading to 
```
"react-router": "6.0.0-beta.0",
"react-router-dom": "6.0.0-beta.0",
```
indeed produces the error.

Hard to say why, but unless anyone has a valid use case for staying on the beta versions, I agree with closing this issue with the solution being to upgrade `react-router` and `react-router-dom`.

Big thanks to everyone for quick responses! 🥇

**Comment by dmckernanacv** [comment]
I would like to add that there may be a bug with the version:bump cli then, as we've upgraded every minor version available and it somehow missed those dependencies (I'm assuming the others faced this as well?)

**Comment by Rugvip** [comment]
@dmckernanacv the `version:bump` command only bumps `@backstage/*` packages by default, although you can ask it to bump other packages too. The `react-router` one in particular is something we didn't want to force people to do, but rather have it be a migration that you can do at your own pace.

@agentbellnorm as mentioned ^, we're still looking to support React Router v6 beta for now. We'll most likely keep supporting it in the core until v2, but that doesn't mean that all plugins need to support it. Still, I think it might be worth a small amount of effort to see if we can keep support for beta still.

**Comment by agentbellnorm** [comment]
#15312 Adds backwards compatibility with `6.0.0-beta.0` 👍
